### PR TITLE
refactor: prefer typed events in the examples

### PR DESCRIPTION
**Pull request by tomivirkki**
Many of the example use an untyped `CustomEvent` that results in the `detail` appearing as `any`

### CODE REVIEW

File: frontend/demo/component/accordion/accordion-summary.ts [hunk 1]
```diff
@@ -50,7 +50,8 @@ export class Example extends LitElement {
       <!-- tag::snippet[] -->
       <vaadin-accordion
         .opened="${this.openedPanelIndex}"
-        @opened-changed="${(e: CustomEvent) => (this.openedPanelIndex = e.detail.value)}"
+        @opened-changed="${(e: CustomEvent<{ value: number }>) =>
```
Comment (web-padawan) [inline frontend/demo/component/accordion/accordion-summary.ts:53]: Consider using `AccordionOpenedChangedEvent` instead.

File: frontend/demo/component/button/button-grid.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ import '@vaadin/vertical-layout';
 import { applyTheme } from 'Frontend/generated/theme';
 import { getPeople } from 'Frontend/demo/domain/DataService';
 import Person from 'Frontend/generated/com/vaadin/demo/domain/Person';
+import { GridSelectedItemsChangedEvent } from '@vaadin/grid/src/vaadin-grid';
```
Comment (web-padawan) [inline frontend/demo/component/button/button-grid.ts:13]:
```suggestion
import type { GridSelectedItemsChangedEvent } from '@vaadin/grid';
```

File: frontend/demo/component/button/button-grid.ts [hunk 3]
```diff
@@ -42,12 +43,12 @@ export class Example extends LitElement {
...
           .items="${this.items}"
-          @selected-items-changed="${(ev: any) =>
-            (this.selectedItems = ev.target ? [...ev.target.selectedItems] : this.selectedItems)}"
+          @selected-items-changed="${(ev: GridSelectedItemsChangedEvent<Person>) =>
+            (this.selectedItems = ev.target ? [...ev.detail.value] : this.selectedItems)}"
         >
           <vaadin-grid-selection-column
             auto-select
-            @select-all-changed="${(ev: CustomEvent) =>
+            @select-all-changed="${(ev: CustomEvent<{ value: boolean }>) =>
```
Comment (web-padawan) [inline frontend/demo/component/button/button-grid.ts:51]: Consider using `GridSelectionColumnSelectAllChangedEvent` (it's quite long though).

File: frontend/demo/component/checkbox/checkbox-group-basic.ts [hunk 4]
```diff
@@ -24,7 +24,7 @@ export class Example extends LitElement {
       <vaadin-checkbox-group
         label="Export data"
         .value="${this.value}"
-        @value-changed="${(e: CustomEvent) => (this.value = e.detail.value)}"
+        @value-changed="${(e: CustomEvent<{ value: string[] }>) => (this.value = e.detail.value)}"
```
Comment (web-padawan) [inline frontend/demo/component/checkbox/checkbox-group-basic.ts:27]: Consider using `CheckboxGroupValueChangedEvent`.

File: frontend/demo/component/checkbox/checkbox-indeterminate.ts [hunk 5]
```diff
@@ -49,7 +49,8 @@ export class Example extends LitElement {
           label="Users to notify"
           theme="vertical"
           .value="${this.selectedIds}"
-          @value-changed="${(e: CustomEvent) => (this.selectedIds = e.detail.value)}"
+          @value-changed="${(e: CustomEvent<{ value: string[] }>) =>
```
Comment (web-padawan) [inline frontend/demo/component/checkbox/checkbox-indeterminate.ts:52]: Consider using `CheckboxGroupValueChangedEvent`.

File: frontend/demo/component/combobox/combo-box-filtering-2.ts [hunk 6]
```diff
@@ -39,8 +39,8 @@ export class Example extends LitElement {
     `;
   }
 
-  private filterChanged(e: CustomEvent) {
-    const filter = e.detail.value as string;
+  private filterChanged(e: CustomEvent<{ value: string }>) {
```
Comment (web-padawan) [inline frontend/demo/component/combobox/combo-box-filtering-2.ts:42]: Consider using `ComboBoxFilterChangedEvent`.

File: frontend/demo/component/combobox/combo-box-presentation.ts [hunk 7]
```diff
@@ -47,8 +47,8 @@ export class Example extends LitElement {
     `;
   }
 
-  private filterChanged(e: CustomEvent) {
-    const filter = e.detail.value as string;
+  private filterChanged(e: CustomEvent<{ value: string }>) {
```
Comment (web-padawan) [inline frontend/demo/component/combobox/combo-box-presentation.ts:50]: Consider using `ComboBoxFilterChangedEvent`.

File: frontend/demo/component/confirmdialog/confirm-dialog-basic.ts [hunk 8]
```diff
@@ -54,7 +54,7 @@ export class Example extends LitElement {
     `;
   }
 
-  openedChanged(e: CustomEvent) {
+  openedChanged(e: CustomEvent<{ value: boolean }>) {
```
Comment (web-padawan) [inline frontend/demo/component/confirmdialog/confirm-dialog-basic.ts:57]: Consider using `ConfirmDialogOpenedChangedEvent`.

File: frontend/demo/component/confirmdialog/confirm-dialog-cancel-button.ts [hunk 9]
```diff
@@ -51,7 +51,7 @@ export class Example extends LitElement {
     `;
   }
 
-  openedChanged(e: CustomEvent) {
+  openedChanged(e: CustomEvent<{ value: boolean }>) {
```
Comment (web-padawan) [inline frontend/demo/component/confirmdialog/confirm-dialog-cancel-button.ts:54]: Consider using `ConfirmDialogOpenedChangedEvent`.

File: frontend/demo/component/confirmdialog/confirm-dialog-confirm-button.ts [hunk 10]
```diff
@@ -49,7 +49,7 @@ export class Example extends LitElement {
     `;
   }
 
-  openedChanged(e: CustomEvent) {
+  openedChanged(e: CustomEvent<{ value: boolean }>) {
```
Comment (web-padawan) [inline frontend/demo/component/confirmdialog/confirm-dialog-confirm-button.ts:52]: Consider using `ConfirmDialogOpenedChangedEvent`.

File: frontend/demo/component/confirmdialog/confirm-dialog-reject-button.ts [hunk 11]
```diff
@@ -53,7 +53,7 @@ export class Example extends LitElement {
     `;
   }
 
-  openedChanged(e: CustomEvent) {
+  openedChanged(e: CustomEvent<{ value: boolean }>) {
```
Comment (web-padawan) [inline frontend/demo/component/confirmdialog/confirm-dialog-reject-button.ts:56]: Consider using `ConfirmDialogOpenedChangedEvent`.

File: frontend/demo/component/datetimepicker/date-time-picker-range.ts [hunk 12]
```diff
@@ -30,13 +30,16 @@ export class Example extends LitElement {
         <vaadin-date-time-picker
           label="Start date and time"
           .value="${this.startDateTime}"
-          @value-changed="${(e: CustomEvent) => (this.startDateTime = e.detail.value)}"
+          @value-changed="${(e: CustomEvent<{ value: string }>) =>
```
Comment (web-padawan) [inline frontend/demo/component/datetimepicker/date-time-picker-range.ts:33]: Consider using `DateTimePickerValueChangedEvent`.

File: frontend/demo/component/datetimepicker/date-time-picker-range.ts [hunk 13]
```diff
@@ -30,13 +30,16 @@ export class Example extends LitElement {
...
+            (this.startDateTime = e.detail.value)}"
         ></vaadin-date-time-picker>
+
         <vaadin-date-time-picker
           label="End date and time"
           .min="${this.startDateTime}"
           .value="${this.endDateTime}"
-          @value-changed="${(e: CustomEvent) => (this.endDateTime = e.detail.value)}"
+          @value-changed="${(e: CustomEvent<{ value: string }>) =>
```
Comment (web-padawan) [inline frontend/demo/component/datetimepicker/date-time-picker-range.ts:41]: Consider using `DateTimePickerValueChangedEvent`.

File: frontend/demo/component/dialog/dialog-basic.ts [hunk 14]
```diff
@@ -29,7 +29,8 @@ export class Example extends LitElement {
       <vaadin-dialog
         header-title="New employee"
         .opened="${this.dialogOpened}"
-        @opened-changed="${(e: CustomEvent) => (this.dialogOpened = e.detail.value)}"
+        @opened-changed="${(e: CustomEvent<{ value: boolean }>) =>
```
Comment (web-padawan) [inline frontend/demo/component/dialog/dialog-basic.ts:32]: Consider using `DialogOpenedChangedEvent`.

File: frontend/demo/component/dialog/dialog-closing.ts [hunk 15]
```diff
@@ -28,7 +28,8 @@ export class Example extends LitElement {
       <vaadin-dialog
         aria-label="System maintenance notice"
         .opened="${this.dialogOpened}"
-        @opened-changed="${(e: CustomEvent) => (this.dialogOpened = e.detail.value)}"
+        @opened-changed="${(e: CustomEvent<{ value: boolean }>) =>
```
Comment (web-padawan) [inline frontend/demo/component/dialog/dialog-closing.ts:31]: Consider using `DialogOpenedChangedEvent`.

File: frontend/demo/component/dialog/dialog-draggable.ts [hunk 16]
```diff
@@ -32,7 +32,8 @@ export class Example extends LitElement {
         draggable
         modeless
         .opened="${this.dialogOpened}"
-        @opened-changed="${(e: CustomEvent) => (this.dialogOpened = e.detail.value)}"
+        @opened-changed="${(e: CustomEvent<{ value: boolean }>) =>
```
Comment (web-padawan) [inline frontend/demo/component/dialog/dialog-draggable.ts:35]: Consider using `DialogOpenedChangedEvent`.

File: frontend/demo/component/dialog/dialog-footer.ts [hunk 17]
```diff
@@ -37,7 +37,8 @@ export class Example extends LitElement {
       <vaadin-dialog
         header-title="${`Delete user "${this.user?.firstName} ${this.user?.lastName}"?`}"
         .opened="${this.dialogOpened}"
-        @opened-changed="${(e: CustomEvent) => (this.dialogOpened = e.detail.value)}"
+        @opened-changed="${(e: CustomEvent<{ value: boolean }>) =>
```
Comment (web-padawan) [inline frontend/demo/component/dialog/dialog-footer.ts:40]: Consider using `DialogOpenedChangedEvent`.

File: frontend/demo/component/dialog/dialog-header.ts [hunk 18]
```diff
@@ -42,7 +42,8 @@ export class Example extends LitElement {
       <vaadin-dialog
         header-title="User details"
         .opened="${this.dialogOpened}"
-        @opened-changed="${(e: CustomEvent) => (this.dialogOpened = e.detail.value)}"
+        @opened-changed="${(e: CustomEvent<{ value: boolean }>) =>
```
Comment (web-padawan) [inline frontend/demo/component/dialog/dialog-header.ts:45]: Consider using `DialogOpenedChangedEvent`.

File: frontend/demo/component/dialog/dialog-no-padding.ts [hunk 19]
```diff
@@ -42,7 +42,8 @@ export class Example extends LitElement {
         theme="no-padding"
         header-title="Filter reports by users:"
         .opened="${this.dialogOpened}"
-        @opened-changed="${(e: CustomEvent) => (this.dialogOpened = e.detail.value)}"
+        @opened-changed="${(e: CustomEvent<{ value: boolean }>) =>
```
Comment (web-padawan) [inline frontend/demo/component/dialog/dialog-no-padding.ts:45]: Consider using `DialogOpenedChangedEvent`.

File: frontend/demo/component/inputfields/input-field-helper.ts [hunk 20]
```diff
@@ -66,7 +66,7 @@ export class Example extends LitElement {
     `;
   }
 
-  private onPasswordChanged(e: CustomEvent) {
+  private onPasswordChanged(e: CustomEvent<{ value: string }>) {
```
Comment (web-padawan) [inline frontend/demo/component/inputfields/input-field-helper.ts:69]: Consider using `PasswordFieldValueChangedEvent`.

File: frontend/demo/component/passwordfield/password-field-advanced-helper.ts [hunk 21]
```diff
@@ -58,7 +58,7 @@ export class Example extends LitElement {
     `;
   }
 
-  private onPasswordChanged(e: CustomEvent) {
+  private onPasswordChanged(e: CustomEvent<{ value: string }>) {
```
Comment (web-padawan) [inline frontend/demo/component/passwordfield/password-field-advanced-helper.ts:61]: Consider using `PasswordFieldValueChangedEvent`.

File: frontend/demo/component/radiobutton/radio-button-custom-option.ts [hunk 22]
```diff
@@ -38,7 +38,7 @@ export class Example extends LitElement {
           label="Payment method"
           theme="vertical"
           .value="${this.value}"
-          @value-changed="${(e: CustomEvent) => (this.value = e.detail.value)}"
+          @value-changed="${(e: CustomEvent<{ value: string }>) => (this.value = e.detail.value)}"
```
Comment (web-padawan) [inline frontend/demo/component/radiobutton/radio-button-custom-option.ts:41]: Consider using `RadioGroupValueChangedEvent`.

File: frontend/demo/component/textarea/text-area-basic.ts [hunk 23]
```diff
@@ -26,7 +26,7 @@ export class Example extends LitElement {
         label="Comment"
         .maxlength="${this.charLimit}"
         .value="${this.text}"
-        @value-changed="${(e: CustomEvent) => (this.text = e.detail.value)}"
+        @value-changed="${(e: CustomEvent<{ value: string }>) => (this.text = e.detail.value)}"
```
Comment (web-padawan) [inline frontend/demo/component/textarea/text-area-basic.ts:29]: Consider using `TextAreaValueChangedEvent`.

File: frontend/demo/component/textarea/text-area-helper.ts [hunk 24]
```diff
@@ -35,7 +35,7 @@ export class Example extends LitElement {
         label="Description"
         .maxlength="${this.charLimit}"
         .value="${this.text}"
-        @value-changed="${(e: CustomEvent) => (this.text = e.detail.value)}"
+        @value-changed="${(e: CustomEvent<{ value: string }>) => (this.text = e.detail.value)}"
```
Comment (web-padawan) [inline frontend/demo/component/textarea/text-area-helper.ts:38]: Consider using `TextAreaValueChangedEvent`.

File: frontend/demo/component/accordion/accordion-summary.ts (hunk 1 above)
Comment (tomivirkki) [reply frontend/demo/component/accordion/accordion-summary.ts]: Done. Didn't realize we had dedicated types also for events that use `notify` 👍

**Review by web-padawan — APPROVED** [review]

**Pull request merged**